### ISSUE TITLE
Implement wholeListMerge, omitWhenMerged, versionId

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/ocds-merge#2]
http://standard.open-contracting.org/latest/en/schema/merging/

I can't find where this repo used `mergeStrategy` or `mergeOptions` from 1.0… I guess that was only in https://github.com/open-contracting-archive/jsonmerge

In fact, I can't find where this repo used the schema as a guide for merging…

**Comment by kindly** [comment]
@jpmckinney yes this library hardcoded "wholeListMerge" and "omitWhenMerged" values and implied "versionId" values.
Discussion about these rules are here: 
https://github.com/open-contracting/standard/issues/287

This was mainly due to the transition between 1.0 and 1.1 to make sure the library will work when we upgraded and not need to rely on the old merge strategies.

However, clearly this will not work for extensions in 1.1 and the code could be simpler if we used these properties.  

So I think we should save the current code in a branch and call it 1.0 and maybe add this to pypi as a different library or an old version.

Then we can use these properties now, for an updated version of this library as a proper reference implementation, and therefore document it better also.

Also I think a javascript version would be useful. There is some work started on this already.
https://gist.github.com/kindly/77877c6c8fee9028e797d2442b223d51
